File: src/main/java/io/github/oleksivio/telegram/bot/api/model/method/sticker/GetStickerSet.java
package io.github.oleksivio.telegram.bot.api.model.method.sticker;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.oleksivio.telegram.bot.api.model.NetworkError;
import io.github.oleksivio.telegram.bot.core.controller.network.ActionNetworker;
import io.github.oleksivio.telegram.bot.core.model.method.RunnableAction;
import io.github.oleksivio.telegram.bot.core.model.responses.CommonResponse;
import io.github.oleksivio.telegram.bot.core.model.responses.StickerSetResponse;
import io.github.oleksivio.telegram.bot.api.model.objects.std.sticker.StickerSet;

/**
 * @see <a href="https://core.telegram.org/bots/api#getstickerset>getStickerSet</a>
 */
public class GetStickerSet extends RunnableAction<StickerSet> {
    private static final String METHOD = "getStickerSet";

    /**
     * name String Yes Name of the sticker set
     */
    @JsonProperty("name")
    private String name = null;

    public GetStickerSet(ActionNetworker actionNetworker) {
        super(METHOD, actionNetworker);
    }

    public String getName() {
        return name;
    }

    public GetStickerSet setName(String name) {
        this.name = name;
        return this;
    }

    @Override
    protected Class<? extends CommonResponse<StickerSet>> getResultWrapperClass() {
        return StickerSetResponse.class;
    }

    @Override
    public GetStickerSet setNetworkErrorListener(NetworkError onNetworkError) {
        pSetNetworkErrorListener(onNetworkError);
        return this;
    }
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/model/method/sticker/GetStickerSet.kt
package io.github.oleksivio.telegram.bot.api.model.method.sticker

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.telegram.bot.api.model.objects.std.sticker.StickerSet
import io.github.oleksivio.telegram.bot.core.model.ApiDict
import io.github.oleksivio.telegram.bot.core.model.StickerSetResponse
import io.github.oleksivio.telegram.bot.core.model.method.Action

/**
 * @see [getStickerSet](https://core.telegram.org/bots/api/#getstickerset)
 */
data class GetStickerSet(
        /**
         * name String Yes Name of the sticker set
         */
        @JsonProperty(ApiDict.NAME_KEY)
        val name: String
) : Action<StickerSet>() {

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "getStickerSet"

    @JsonIgnore
    override val resultWrapperClass = StickerSetResponse::class
}